signwriting_animation/scripts/minimal_loop.py

```diff
@@ -0,0 +1,265 @@
+        self.val_dtws.append(dtw_val.item())
+        self.val_losses.append(loss.item())
+        self.log("val/loss", loss, prog_bar=True)
+
+    def on_fit_end(self):
+        os.makedirs(self.log_dir, exist_ok=True)
+        csv_path = os.path.join(self.log_dir, "minimal_metrics.csv")
+        with open(csv_path, "w", newline="") as f:
+            w = csv.writer(f)
+            w.writerow(["step", "train_loss", "val_loss", "val_dtw"])
+            max_len = max(len(self.train_losses), len(self.val_losses), len(self.val_dtws))
+            for i in range(max_len):
+                tr  = self.train_losses[i] if i < len(self.train_losses) else ""
+                vl  = self.val_losses[i]  if i < len(self.val_losses)  else ""
+                dtw = self.val_dtws[i]    if i < len(self.val_dtws)    else ""
+                w.writerow([i + 1, tr, vl, dtw])
+
+        import matplotlib.pyplot as plt
+        plt.figure()
+        if self.train_losses:
+            plt.plot(self.train_losses, label="train_loss")
+        if self.val_losses:
+            plt.plot(self.val_losses, label="val_loss")
+        if self.val_dtws:
+            plt.plot(self.val_dtws, label="val_dtw")
+        plt.xlabel("steps")
+        plt.legend()
+        plt.tight_layout()
+        plt.savefig(os.path.join(self.log_dir, "minimal_curves.png"))
+        plt.close()
+
+    def configure_optimizers(self):
+        return torch.optim.AdamW(self.parameters(), lr=self.lr)
+
+
+def make_loader(data_dir, csv_path, split, bs, num_workers):
+    """
+    Build a DataLoader with:
+    - Dataset returning MaskedTensor.
+    - zero_pad_collator to align sequences along time dimension.
+    - FilteredDataset to pick a small, valid subset for a minimal run.
+    """
+    base = DynamicPosePredictionDataset(data_dir=data_dir, csv_path=csv_path, with_metadata=True, split=split)
+    ds = FilteredDataset(base, target_count=200, max_scan=3000)
+    print(f"[DEBUG] split={split} | batch_size={bs} | len(ds)={len(ds)}")
+    return DataLoader(ds, batch_size=bs, shuffle=False, num_workers=num_workers, collate_fn=zero_pad_collator)
+
+
+if __name__ == "__main__":
+    pl.seed_everything(42, workers=True)
+    torch.set_default_dtype(torch.float32)
+
+    data_dir = "/data/yayun/pose_data"
+    csv_path = "/data/yayun/signwriting-animation/data.csv"
+
+    batch_size = 4
+    num_workers = 2
+
+    train_loader = make_loader(data_dir, csv_path, "train", bs=batch_size, num_workers=num_workers)
+    val_loader = train_loader
+
+    model = LitMinimal(log_dir="logs")
+
+    trainer = pl.Trainer(
+        max_steps=1000,
+        accelerator="gpu" if torch.cuda.is_available() else "cpu",
+        devices=1,
+        log_every_n_steps=5,
+        limit_train_batches=10,
+        limit_val_batches=5,
+        check_val_every_n_epoch=1,
+        enable_checkpointing=False,
+        deterministic=True,
+        num_sanity_val_steps=0,
+    )
+    trainer.fit(model, train_loader, val_loader)
```

Comment:
ok, let's say it fits. now what? how can we validate that it is correct? that the model can reproduce a single training example? you can train on like 4 examples from the dataset (the loss will go down fast), and then take these samples, and run the inference loop - given the first frame, predict the next `future_frames` etc. then you will see with your eyes if the model is able to overfit or not